test/base/UpgradeableProxyUpgradeToTest.js

```diff
@@ -1,139 +1,80 @@
 // JS Libraries
-const { upgradeable } = require("../utils/events");
 const withData = require("leche").withData;
-const { t } = require("../utils/consts");
+const { t, NULL_ADDRESS } = require("../utils/consts");
+const {
+    upgradeable
+} = require('../utils/events');
 
 // Mock contracts
 const Mock = artifacts.require("./mock/util/Mock.sol");
 
 // Smart contracts
 const UpgradeableProxy = artifacts.require("./base/UpgradeableProxy.sol");
-const UpgradableV1 = artifacts.require("./mock/upgradable/UpgradableV1.sol");
-const UpgradableV2 = artifacts.require("./mock/upgradable/UpgradableV2.sol");
-
-const DAI = artifacts.require("./mock/token/DAIMock.sol");
+const Settings = artifacts.require("./base/Settings.sol");
+const { createTestSettingsInstance } = require("../utils/settings-helper");
 
 contract("UpgradeableProxyUpgradeToTest", function(accounts) {
-    let dai;
-    let v1LibraryInstance;
-    let v2LibraryInstance;
-    let v1InitData;
-    let initValue = 43;
-    let settings;
-
-    before(async () => {
-        settings = await Mock.new();
-        dai = await DAI.new()
-    })
+    let settingsLogic;
+    let initLogic;
 
     beforeEach("Setup for each test", async () => {
-        
-        v1LibraryInstance = await UpgradableV1.new();
-        await v1LibraryInstance.initialize(initValue);
-        v2LibraryInstance = await UpgradableV2.new();
-        await v2LibraryInstance.initialize(initValue);
-
-        v1InitData = v1LibraryInstance.contract.methods.initialize(initValue).encodeABI();
+        const owner = accounts[0];
+        settingsLogic = await createTestSettingsInstance(
```

Comment:
since this is being used here as the actual settings contract and not the logic, lets rename to just `settings`

Reply:
Renamed